app-frontend/src/app/pages/lab/run/run.controller.js

```diff
@@ -190,9 +355,30 @@ export default class LabRunController {
     createToolRun() {
         this.toolService.createToolRun(this.toolRun).then(tr => {
             this.lastToolRun = tr;
+            this.clearWarning();
```

Comment:
I would clear the warning optimistically (do it before the request) and then set the warning if necessary. It would add some snappiness to the UI that indicated pressing the 'Apply' button did something.